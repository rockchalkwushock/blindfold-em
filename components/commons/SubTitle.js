import PropTypes from 'prop-types'

/**
 * @function SubTitle
 * @description renders <h2 />
 *
 * @prop {String} text
 * @returns React Element
 */
const SubTitle = ({ text }) => <h2>{text}</h2>

SubTitle.propTypes = {
  text: PropTypes.string.isRequired
}

export default SubTitle
