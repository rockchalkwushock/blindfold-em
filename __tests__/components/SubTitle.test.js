import React from 'react'
import { render } from 'enzyme'

import SubTitle from '../../components/commons/SubTitle'

const Test = () => <SubTitle text="What activity are you working on?" />

describe('Component: <SubTitle />', () => {
  test('should render matching text for <h2/>', () => {
    const tree = render(<Test />)
    expect(tree.text()).toEqual('What activity are you working on?')
  })

  test('should render without exploding', () => {
    const tree = render(<Test />)
    expect(tree).toMatchSnapshot()
  })
})
