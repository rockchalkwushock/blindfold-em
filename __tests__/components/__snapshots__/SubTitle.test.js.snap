// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <SubTitle /> should render without exploding 1`] = `
<h2>
  What activity are you working on?
</h2>
`;
